core/node/da_clients/src/eigen/verifier.rs

```diff
@@ -0,0 +1,859 @@
+use std::{collections::HashMap, str::FromStr};
+
+use ark_bn254::{Fq, G1Affine};
+use ethabi::{encode, Token};
+use rust_kzg_bn254::{blob::Blob, kzg::Kzg, polynomial::PolynomialFormat};
+use tiny_keccak::{Hasher, Keccak};
+use zksync_basic_types::web3::CallRequest;
+use zksync_eth_client::clients::PKSigningClient;
+use zksync_types::{
+    url::SensitiveUrl,
+    web3::{BlockId, BlockNumber},
+    K256PrivateKey, SLChainId, H160, U256,
+};
+use zksync_web3_decl::client::{Client, DynClient, L1};
+
+use super::blob_info::{BatchHeader, BlobHeader, BlobInfo, G1Commitment};
+
+#[derive(Debug)]
+pub enum VerificationError {
+    ServiceManagerError,
+    WrongUrl,
+    KzgError,
+    WrongProof,
+    DifferentCommitments,
+    DifferentRoots,
+    EmptyHash,
+    DifferentHashes,
+    WrongQuorumParams,
+    QuorumNotConfirmed,
+    CommitmentNotOnCurve,
+    CommitmentNotOnCorrectSubgroup,
+}
+
+/// Configuration for the verifier used for authenticated dispersals
+#[derive(Debug, Clone)]
+pub struct VerifierConfig {
+    pub verify_certs: bool,
+    pub rpc_url: String,
+    pub svc_manager_addr: String,
+    pub max_blob_size: u32,
+    pub path_to_points: String,
+    pub eth_confirmation_depth: u32,
+    pub private_key: String,
+    pub chain_id: u64,
+}
+
+/// Verifier used to verify the integrity of the blob info
+/// Kzg is used for commitment verification
+/// EigenDA service manager is used to connect to the service manager contract
+#[derive(Debug, Clone)]
+pub struct Verifier {
+    kzg: Kzg,
+    cfg: VerifierConfig,
+    signing_client: PKSigningClient,
+}
+
+impl Verifier {
+    const DEFAULT_PRIORITY_FEE_PER_GAS: u64 = 100;
+    const BATCH_ID_TO_METADATA_HASH_FUNCTION_SELECTOR: [u8; 4] = [236, 203, 191, 201];
```

Comment:
this is going to be difficult to maintain, could you please use something like [this](https://github.com/matter-labs/zksync-era/blob/main/core/lib/contracts/src/lib.rs)?

Reply:
just recalled that there is an easier way where you don't need to include the contracts in our codebase, example:
```       
let func_selector = ethabi::short_signature(
            "validatePubdata",
            &[
                ParamType::FixedBytes(32),
                ParamType::FixedBytes(32),
                ParamType::FixedBytes(32),
                ParamType::FixedBytes(32),
                ParamType::Bytes,
            ],
        );
```